### PR TITLE
chore(ci): start using mill-ci-release

### PR DESCRIPTION
**Pull request by ckipp01**
No functionality here should change. The only difference you'll notice
is that you'll not be publishing SNAPSHOT versions on merge to main.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+          distribution: adopt
+          java-version: 11
+
+    - name: Setup Mill
+      uses: jodersky/setup-mill@master
+      with:
+        mill-version: 0.10.7
+
+    - name: Publish
+      run: mill -j 0 -k -i --disable-ticker io.kipp.mill.ci.release.ReleaseModule/publishAll
```
Comment (ckipp01) [inline .github/workflows/release.yml:30]: I feel like I see this different for every CI I look at. I see you weren't using `-i`, how come? I see you use `--disable-ticker`, does that save speed?

File: build.sc [hunk 2]
```diff
@@ -270,12 +270,12 @@ trait RPCCrossPlatformModule extends Module { shared =>
...
 }
 
-trait JsonRPCModule extends ScalaModule with PublishModule with scalafmt.ScalafmtModule {
+trait JsonRPCModule extends ScalaModule with CiReleaseModule with scalafmt.ScalafmtModule {
   def scalafmt() = T.command(reformat())
   def fmt() = T.command(reformat())
   def refreshedEnv = T.input(T.ctx().env)
   def publishVersion = T {
-    if (refreshedEnv().contains("CI")) VcsVersion.vcsState().format()
+    if (refreshedEnv().contains("CI")) super.publishVersion()
```
Comment (ckipp01) [inline build.sc:278]: Under the hood mill-ci-release is using this plus some goodies to add on `-SNAPSHOT`. For now I manually do that until I get the ok to add it https://github.com/lefou/mill-vcs-version/discussions/62.

File: .github/workflows/release.yml (hunk 1 above)
Comment (Baccata) [reply .github/workflows/release.yml:30]: >  I see you weren't using -i, how come?  

just a mistake on my end 

> I see you use --disable-ticker, does that save speed? 

It makes logs less verbose imho. Plus I think @keynmol was the one to implement `--disable-ticker` on mill, and it's important to make him feel like people care about what he builds^^

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -34,36 +33,10 @@ jobs:
...
-        mill-version: 0.10.5
+        mill-version: 0.10.7
 
     - name: Run tests
       run: |
         mill -k --disable-ticker __.resolvedIvyDeps &&
         mill -k --disable-ticker mill.scalalib.scalafmt.ScalafmtModule/checkFormatAll __.sources &&
         mill -j 0 -k --disable-ticker __.test
-
-    - name: Set up GPG
```
Comment (Baccata) [inline .github/workflows/ci.yml:45]: 🎉

File: .github/workflows/release.yml (hunk 1 above)
Comment (keynmol) [reply .github/workflows/release.yml:30]: Yeah it's purely for my ego.

Comment (ckipp01) [reply .github/workflows/release.yml:30]: Damn, I'll start using it immediately then.

**Review by Baccata — APPROVED** [review]

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      env:
+        PGP_PASSPHRASE: ${{ secrets.PGP_PASSPHRASE }}
+        PGP_SECRET: ${{ secrets.PGP_SECRET }}
```
Comment (ckipp01) [inline .github/workflows/release.yml:33]: This failed to publish the snapshot with:

> io.kipp.mill.ci.release.ReleaseModule.setupEnv Missing PGP_SECRET. Make sure you have it set.

🤔 do you see something silly I missed here? I see you had this named the same before, so this should exist.

**Pull request merged**